DEVELOPER.md

```diff
@@ -0,0 +1,167 @@
+
+### Virtual Box setup
+
+You will need to setup port forwarding on VirtualBox to be able to ssh into the box which can be done as follows:
+
+1. Select the virtual machine you want to use and click "Settings"
+1. Click "Network" tab
+1. Click "Advanced" dropdown
+1. Click "Port Forwarding"
+1. Add a new rule in the popup. The only thing you will need is "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)
+1. Click "Ok"
+
+Now you should be able to ssh <your_name>@127.1 -p 22222 to get SSH prompt. However, this requires us to type a long command and password every time we sign in. It is recommended you setup a public key and SSH alias to make this process simpler:
+
+1. On your host machine, generate a keypair for SSH into the guest: `ssh-keygen -t ed25519`.
+Just keep hitting Enter until the key is generated. You do not need a password for this key file since it is only used for SSH into your guest
+1. Type `cat .ssh/id_ed25519.pub` and copy the public key
+1. SSH into the guest using the command above
+1. Create the ssh config directory (if it doesn't exist) `$ mkdir -p .ssh`
+1. Edit the authorized keys list: `vim .ssh/authorized_keys`
+1. Paste in the content of .ssh/id_ed25519.pub
+1. Adjust the required privileges: `chmod 700 .ssh/`  and `chmod 400 .ssh/authorized_keys`
+1. Logout of guest and make sure you are not promoted password when SSH again
+1. Edit the .ssh/config file on your host and put in the following content:
+
+```
+    Host dev
+        HostName 127.1
+        Port 22222
+        User <your_user_name>
+```
+
+Now try ssh dev on your host, you should be able to get into the guest directly
+
+## Linux Environment
+
+Once you have a Linux development environment (either virtual or bare metal), the build is done in four separate steps:
+
+1. Prerequisite packages.  Mostly compilers, tools and libraries needed to compile everything else.
+1. OpenResty system, including Nginx, LuaJIT, PCRE, etc.
+1. Databases. Kong uses Posgres, Cassandra and Redis.  We have a handy setup with docker-compose to keep each on its container.
+1. Kong itself.
+
+
+### Prerequisites
+
+These are the needed tools and libraries that aren't installed out of the box on Ubuntu and Fedora, respectively.  Just run one of these, either as root or sudo.
+
+Ubuntu:
+
+```
+    apt-get update
+
+    apt-get install \
+        build-essential \
+        curl \
+        docker \
+        docker-compose \
+        git \
+        libpcre3 \
+        libyaml-dev \
+        m4 \
+        openssl \
+        perl \
+        procps \
+        unzip \
+        zlib1g-dev
+```
+
+Fedora:
+
+```
+    dnf install \
+        automake \
+        docker \
+        docker-compose \
+        gcc \
+        gcc-c++ \
+        git \
+        libyaml-devel \
+        make \
+        patch \
+        pcre-devel \
+        unzip \
+        zlib-devel
+```
```

Comment:
I've tried the command from a fresh Debian image: we should include autoconf, libyaml-dev, g++, curl as well.

Reply:
libyaml-dev and curl are listed specifically in the `apt install` line; g++ is included by the build-essential metapackage, I'm pretty sure autoconf was installed too, but better to add specifically